File: ports/csharp/src/gui/kernel/BcsWindow.cs
using BCS.Core.Tools;

namespace BCS.Gui.Kernel {
    // BcsWindow represents a top-level native window
    public class BcsWindow {
        public string Title { get; set; }
        public BcsRect Geometry { get; set; }
        public bool IsVisible { get; private set; }

        public BcsWindow() {
            Title = string.Empty;
            Geometry = new BcsRect(0, 0, 800, 600);
            IsVisible = false;
        }

        public void SetGeometry(int x, int y, int width, int height) {
            Geometry = new BcsRect(x, y, width, height);
        }

        public void Show() {
            IsVisible = true;
        }

        public void Hide() {
            IsVisible = false;
        }
    }
}

File: ports/java/src/gui/kernel/BcsWindow.java
package bcs.gui.kernel;

import bcs.core.tools.BcsGeometry.BcsRect;

// BcsWindow represents a top-level native window
public class BcsWindow {
    private String title;
    private BcsRect geometry;
    private boolean visible;

    public BcsWindow() {
        this.title = "";
        this.geometry = new BcsRect(0, 0, 800, 600);
        this.visible = false;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setGeometry(int x, int y, int width, int height) {
        this.geometry = new BcsRect(x, y, width, height);
    }

    public void show() {
        this.visible = true;
    }

    public void hide() {
        this.visible = false;
    }
}
